src/migration/1647526647810-init.ts

```diff
@@ -0,0 +1,116 @@
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`CREATE TABLE "iasql_module" ("name" character varying NOT NULL, CONSTRAINT "UQ_e91a0b0e9a029428405fdd17ee4" UNIQUE ("name"), CONSTRAINT "PK_e91a0b0e9a029428405fdd17ee4" PRIMARY KEY ("name"))`);
+        await queryRunner.query(`CREATE TABLE "iasql_tables" ("table" character varying NOT NULL, "module" character varying NOT NULL, CONSTRAINT "PK_2e2832f9cf90115571eb803a943" PRIMARY KEY ("table", "module"))`);
+        await queryRunner.query(`CREATE TYPE "public"."iasql_operation_optype_enum" AS ENUM('APPLY', 'SYNC', 'INSTALL', 'UNINSTALL')`);
+        await queryRunner.query(`CREATE TABLE "iasql_operation" ("opid" uuid NOT NULL, "start_date" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), "end_date" TIMESTAMP WITH TIME ZONE, "optype" "public"."iasql_operation_optype_enum" NOT NULL, "params" text array NOT NULL, "output" text, "err" text, CONSTRAINT "PK_edf11c327fef1bf78dd04cdf3ce" PRIMARY KEY ("opid"))`);
+        await queryRunner.query(`CREATE TABLE "iasql_dependencies" ("module" character varying NOT NULL, "dependency" character varying NOT NULL, CONSTRAINT "PK_b07797cfc364fa84b6da165f89d" PRIMARY KEY ("module", "dependency"))`);
+        await queryRunner.query(`CREATE INDEX "IDX_9732df6d7dff34b6f6a1732033" ON "iasql_dependencies" ("module") `);
+        await queryRunner.query(`CREATE INDEX "IDX_7dbdaef2c45fdd0d1d82cc9568" ON "iasql_dependencies" ("dependency") `);
+        await queryRunner.query(`ALTER TABLE "iasql_tables" ADD CONSTRAINT "FK_0e0f2a4ef99e93cfcb935c060cb" FOREIGN KEY ("module") REFERENCES "iasql_module"("name") ON DELETE NO ACTION ON UPDATE NO ACTION`);
+        await queryRunner.query(`ALTER TABLE "iasql_dependencies" ADD CONSTRAINT "FK_9732df6d7dff34b6f6a1732033b" FOREIGN KEY ("module") REFERENCES "iasql_module"("name") ON DELETE CASCADE ON UPDATE CASCADE`);
+        await queryRunner.query(`ALTER TABLE "iasql_dependencies" ADD CONSTRAINT "FK_7dbdaef2c45fdd0d1d82cc9568c" FOREIGN KEY ("dependency") REFERENCES "iasql_module"("name") ON DELETE CASCADE ON UPDATE CASCADE`);
+        await queryRunner.query(`
+            create or replace procedure until_iasql_operation(_optype text, _params text[])
+            language plpgsql
+            as $$
+            declare
+                _opid uuid;
+                _counter integer := 0;
+                _output text;
+                _err text;
+            begin
+                select md5(random()::text || clock_timestamp()::text)::uuid into _opid;
+                insert into iasql_operation (opid, optype, params)
+                values (_opid, _optype::iasql_operation_optype_enum, _params);
+                -- schedule job
+                perform graphile_worker.add_job('operation', json_build_object('opid', _opid, 'optype', _optype, 'params', _params));
+                -- commit transaction so the worker can see the row
+                commit;
```

Comment:
beyond having to level up my sql skills, this part took me a bit to figure out. without this the procedure call always timed out

Reply:
no, the stored procedure creates a transaction so you have to tell it to commit what has been done so far

Reply:
so basically without this `commit` the row in `iasql_operation` doesn't exist until the SP returns

Reply:
oh I see! yes, the complete sp runs in transaction. Makes sense, I know it also exists something call saveppoints I think, not sure how they work but I think they are used in these situations

Reply:
oh interesting, I didn't know about save points